native/src/main/java/io/ballerina/lib/data/xmldata/utils/DataUtils.java

```diff
@@ -782,6 +783,27 @@ private static String addAttributeToRecord(BString prefix, BString uri, String k
         return prefix.getValue().concat(Constants.COLON).concat(key);
     }
 
+    public static Object validateConstraints(Object convertedValue, BTypedesc typed, boolean requireValidation) {
+        if (!requireValidation) {
+            return convertedValue;
```

Comment:
we need to add a test with disabling the validation

Reply:
Updated https://github.com/ballerina-platform/module-ballerina-data.xmldata/pull/21/commits/0bbdaa6abae99ebea2102107811d1ad707bc051e